Tests/Functional/Result/PersistObjectsTest.php

```diff
@@ -91,4 +92,33 @@ public function testAppendObject()
 
         $this->count(3, $product->relatedCategories);
     }
+
+    /**
+     * Tests the type of int and float parameters retrieved from elasticsearch
+     * when objects are created with strings
+     */
+    public function testType()
+    {
+        $manager = $this->getManager();
+
+        $product = new Product();
+        $product->id = 'foo';
+        $product->title = 'Test Document';
+        $product->price = '8.95';
```

Comment:
What if a float value is 8,95 (comma instead a doc). This depends on locale settings, but still. @mvar do you think it's worth to consider?